karrot/activities/tests/test_renderers.py

```diff
@@ -0,0 +1,40 @@
+from django.test import TestCase
+from pytz import UTC
+from datetime import datetime
+from rest_framework.utils.serializer_helpers import ReturnDict, ReturnList
+from rest_framework.response import Response
+from karrot.activities.renderers import ICSCalendarRenderer
+from icalendar import vCalAddress, vText
+
+
+class ICSCalendarRendererTest(TestCase):
+    def setUp(self):
+        self.renderer = ICSCalendarRenderer()
+        attendee = vCalAddress('MAILTO:marysmith@example.com')
+        attendee.params['cn'] = vText('Mary Smith')
+        attendee.params['ROLE'] = vText('REQ-PARTICIPANT')
+        self.event = ReturnDict({
+            'dtstart': datetime(year=2021, month=3, day=19, hour=17, tzinfo=UTC),
+            'dtend': datetime(year=2021, month=3, day=19, hour=18, tzinfo=UTC),
+            'description': 'Hello\nWorld!',
+            'transp': 'OPAQUE',
+            'attendee': attendee,
+            'toskip': None
+        },
+                                serializer=None)
+        self.expected_output = '\r\n'.join([
+            "BEGIN:VCALENDAR", "VERSION:2.0", "PRODID:-//Karrot//EN", "BEGIN:VEVENT",
+            "DTSTART;VALUE=DATE-TIME:20210319T170000Z", "DTEND;VALUE=DATE-TIME:20210319T180000Z",
+            "ATTENDEE;CN=\"Mary Smith\";ROLE=REQ-PARTICIPANT:MAILTO:marysmith@example.com",
+            "DESCRIPTION:Hello\\nWorld!", "TRANSP:OPAQUE", "END:VEVENT", "END:VCALENDAR"
```

Comment:
minor: yapf will do the formatting nicely here but only if there is a trailing comma